lib/ansible/modules/cloud/google/gcp_dns_managed_zone.py

```diff
@@ -0,0 +1,239 @@
+        required: false
+    dns_name:
+        description:
+            - The DNS name of this managed zone, for instance "example.com.".
+        required: false
+    name:
+        description:
+            - User assigned name for this resource.
+              Must be unique within the project.
+        required: true
+    name_server_set:
+        description:
+            - Optionally specifies the NameServerSet for this ManagedZone. A
+              NameServerSet is a set of DNS name servers that all host the same
+              ManagedZones. Most users will leave this field unset.
+        required: false
+extends_documentation_fragment: gcp
+'''
+
+EXAMPLES = '''
+- name: Create a Managed Zone
+  gcp_dns_managed_zone:
+      name: testObject
+      dns_name: test.somewild2.example.com.
+      description: 'test zone'
+      project: testProject
+      auth_kind: service_account
+      service_account_file: /tmp/auth.pem
+      scopes:
+        - https://www.googleapis.com/auth/ndev.clouddns.readwrite
+      state: 'present'
+'''
+
+RETURNS = '''
+    id:
+        description:
+            - Unique identifier for the resource; defined by the server.
+        returned: success
+        type: integer
+    name_servers:
+        description:
+            - Delegate your managed_zone to these virtual name servers;
+              defined by the server
+        returned: success
+        type: array
+    creation_time:
+        description:
+            - The time that this resource was created on the server.
+              This is in RFC3339 text format.
+        returned: success
+        type: time
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible.module_utils.gcp_utils import navigate_hash, GcpAuthentication
+from ansible.module_utils.basic import AnsibleModule
+import json
+
+################################################################################
+# Main
+################################################################################
+
+
+def main():
+    """Main function"""
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            description=dict(type='str'),
+            dns_name=dict(type='str'),
+            name=dict(required=True, type='str'),
+            name_server_set=dict(type='list'),
+            project=dict(required=True, type='str'),
+            auth_kind=dict(
+                required=True,
+                choices=['machineaccount', 'serviceaccount', 'application'],
+                type='str'),
+            service_account_email=dict(type='str'),
+            service_account_file=dict(type='path'),
+            scopes=dict(required=True, type='list')
+        ),
+        mutually_exclusive=[['service_account_email', 'service_account_file']],
```

Comment:
Worth adding this to the `documentation` fragments for both options, e.g.
`The I(service_account_email) and I(service_account_file)` options are mutually exclusive.`

Reply:
Done.